airflow/contrib/hooks/gcp_dataflow_hook.py

```diff
@@ -31,88 +31,111 @@
+            elif job:
+                self.log.info(
+                    'Google Cloud DataFlow with job_id %s has name %s',
+                    self._job_id, job['name']
+                )
+            else:
+                self.log.info(
+                    'Google Cloud DataFlow job not available yet..'
+                )
 
-        return job
+        return self._jobs
 
     def wait_for_done(self):
         while True:
-            if self._job and 'currentState' in self._job:
-                if 'JOB_STATE_DONE' == self._job['currentState']:
-                    return True
-                elif 'JOB_STATE_RUNNING' == self._job['currentState'] and \
-                     'JOB_TYPE_STREAMING' == self._job['type']:
-                    return True
-                elif 'JOB_STATE_FAILED' == self._job['currentState']:
-                    raise Exception("Google Cloud Dataflow job {} has failed.".format(
-                        self._job['name']))
-                elif 'JOB_STATE_CANCELLED' == self._job['currentState']:
-                    raise Exception("Google Cloud Dataflow job {} was cancelled.".format(
-                        self._job['name']))
-                elif 'JOB_STATE_RUNNING' == self._job['currentState']:
-                    time.sleep(self._poll_sleep)
-                elif 'JOB_STATE_PENDING' == self._job['currentState']:
-                    time.sleep(15)
-                else:
-                    self.log.debug(str(self._job))
-                    raise Exception(
-                        "Google Cloud Dataflow job {} was unknown state: {}".format(
-                            self._job['name'], self._job['currentState']))
+            for job in self._jobs:
+                if job and 'currentState' in job:
+                    if 'JOB_STATE_DONE' == job['currentState']:
+                        # check all jobs are done
+                        count_not_done = 0
+                        for inner_jobs in self._jobs:
+                            if inner_jobs and 'currentState' in job:
+                                if not 'JOB_STATE_DONE' == inner_jobs['currentState']:
+                                    count_not_done += 1
+                        if count_not_done == 0:
+                            return True
+                    elif 'JOB_STATE_RUNNING' == job['currentState'] and \
+                            'JOB_TYPE_STREAMING' == job['type']:
+                        return True
+                    elif 'JOB_STATE_FAILED' == job['currentState']:
+                        raise Exception("Google Cloud Dataflow job {} has failed.".format(
+                            job['name']))
+                    elif 'JOB_STATE_CANCELLED' == job['currentState']:
+                        raise Exception("Google Cloud Dataflow job {} was cancelled.".format(
+                            job['name']))
+                    elif 'JOB_STATE_RUNNING' == job['currentState']:
+                        None
```

Comment:
This is an odd way of doing this.

1. `pass` is more normal than `None` as a statement in python.
2. `elif job['currentState'] not in ('JOB_STATE_RUNNING','JOB_STATE_PENDING')`

Reply:
I prefer `elif job['currentState'] not in {'JOB_STATE_RUNNING','JOB_STATE_PENDING'}` :-)

Reply:
Agree with `pass` and `elif job['currentState'] not in {'JOB_STATE_RUNNING','JOB_STATE_PENDING'}`

Reply:
sure